client/simple.go

```diff
@@ -10,12 +10,15 @@ import (
 
 var (
 	// 503 error type informing rollup to failover to other DA location
-	ErrServiceUnavailable = fmt.Errorf("eigenda service is unavailable")
+	ErrServiceUnavailable = fmt.Errorf("eigenda service is temporarily unavailable")
 )
 
-// TODO: Add support for custom http client option
 type Config struct {
-	URL string
+	URL string // EigenDA proxy RPC URL
+}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
```

Comment:
Is this really needed here? How are you mocking the client? with some mock library?
Can we use https://pkg.go.dev/net/http/httptest#Server.Client instead to simplify the code and not have this interface (don't think its typical to have an interface for an http client in golang, but not 100% sure)

Reply:
its easiest for me to just bind it to an interface for mocking - done this across quite a few projects in the past